mesonbuild/interpreter/compiler.py

```diff
@@ -773,7 +773,11 @@ def preprocess_method(self, args: T.Tuple[T.List['mesonlib.FileOrString']], kwar
         if any(isinstance(s, (build.CustomTarget, build.CustomTargetIndex, build.GeneratedList)) for s in sources):
             FeatureNew.single_use('compiler.preprocess with generated sources', '1.1.0', self.subproject,
                                   location=self.current_node)
-        tg_name = f'preprocessor_{next(self.preprocess_uid)}'
+        tg_counter = next(self.preprocess_uid)
+        if tg_counter > 0:
+            FeatureNew.single_use('compiler.preprocess used multiple times', '1.1.0', self.subproject,
+                                  location=self.current_node)
```

Comment:
Not that easy, and in retrospect that counter might be wrong. The FeatureNew should only be if more than one preprocess in the same directory.

Reply:
Then we need preprocess_uid to be a dictionary keyed off of self.subdir?

Reply:
Yes, and I think that will also improve rebuilds in the case a new preprocess() call is added somewhere which would cause every other calls to be renumbered.

Reply:
Thanks for fixing my mistake :D

Reply:
Alright, take a look now.